PR5197: arg: Allow flags such as --flag=arg as well as --flag arg.

rwmjones opened the pull request:
Allow flags to be followed directly by their argument, separated by an '=' sign.  This is consistent with what GNU getopt_long and many other command line parsing libraries allow.

Fix for the following issue:
http://caml.inria.fr/mantis/view.php?id=5197

rwmjones committed:
arg: Add no_arg and get_arg helper functions.

The no_arg function in this patch is a no-op.  It will do something
useful in the followups.

The get_arg function simple checks the next position on the command
line exists and returns that argument or raises a Arg.Missing.

This patch should introduce no functional change, it is simply code
refactoring.

In particular, this should not change the treatment of Arg.current
(see: http://caml.inria.fr/mantis/view.php?id=5197#c11147)

rwmjones committed:
arg: Allow flags such as --flag=arg as well as --flag arg.

Allow flags to be followed directly by their argument, separated by an '='
sign.  This is consistent with what GNU getopt_long and many other
command line parsing libraries allow.

Fix for the following issue:
http://caml.inria.fr/mantis/view.php?id=5197

damiendoligez commented:
Fixed bug, added changelog entry and test, merged into trunk (commit 2512b483).

rwmjones commented:
Apparently github never sent me any email notifications so I didn't see the replies here.

Anyway - yes, there were several bugs in this version, which we fixed in the newer versions in the Fedora tree (https://git.fedorahosted.org/cgit/fedora-ocaml.git/log/?h=fedora-24-4.02).  Looking at the comments here and what went into the final OCaml tree, it looks as if you also found and fixed the bugs.
